ktor-utils/common/src/io/ktor/util/pipeline/SuspendFunctionGun.kt

```diff
@@ -19,20 +19,19 @@ internal class SuspendFunctionGun<TSubject : Any, TContext : Any>(
 
     override val coroutineContext: CoroutineContext get() = continuation.context
 
-    
     // this is impossible to inline because of property name clash
     // between PipelineContext.context and Continuation.context
     private val continuation: Continuation<Unit> = object : Continuation<Unit>, CoroutineStackFrame {
         override val callerFrame: CoroutineStackFrame? get() = peekContinuation() as? CoroutineStackFrame
 
-        var currentIndex: Int? = null
+        var currentIndex: Int = Int.MIN_VALUE
 
         override fun getStackTraceElement(): StackTraceElement? = null
 
         private fun peekContinuation(): Continuation<*>? {
-            if (currentIndex == null) currentIndex = lastSuspensionIndex
-            if (currentIndex!! < 0) {
-                currentIndex = null
+            if (currentIndex == Int.MIN_VALUE) currentIndex = lastSuspensionIndex
+            if (currentIndex < 0) {
```

Comment:
When `currentIndex == Int.MIN_VALUE`, execution will go under this branch. Is it ok? If not, we can change the special case to `Int.MAX_VALUE`

Reply:
It's not possible to see `MIN_VALUE` here(see the previous line)